### PR TITLE
Reuse project selector item in renderer to avoid memory leak.

### LINKED ISSUE #1742: Out of memory error when typing and searching multiple projects in Project Selector
Cloud Tools for IntelliJ version: 17.11.2-SNAPSHOT
Google Cloud SDK version: 179.0.0
OS: Linux 4.4.0-97-generic

**What did you do?**
Run -> Edit Configuration, open App Engine Flex deployment configuration (from gs-spring-boot sample code).
Edit project name - type something partial (not a complete project name).
Open project selector drop down to see the accounts/projects tree.
It helps if you have multiple accounts (2 in this scenario) and many available projects in selector tree.

Keep typing and deleting project name. It seems to happen faster if many project names matches the phrase (i.e. "test" in this scenario matched 10+ projects).

**What did you expect to see?**
Filtered list of matching projects.

**What did you see instead?**
Projects match the typed phrase, list expands and collapses as phrase changes for some time.
Then, UI freezes, keys are not accepted.

Memory exceptions are printed to console and continue to be printed when attempting to use the application.

Nov 14, 2017 1:38:51 PM sun.awt.X11.XToolkit processException
WARNING: Exception on Toolkit thread
java.lang.OutOfMemoryError: GC overhead limit exceeded

Exception in thread "JobScheduler FJ pool 8/12" Exception in thread "JobScheduler FJ pool 2/12" java.lang.OutOfMemoryError: Java heap space

### PR DESCRIPTION
**Pull request by ivanporty**
Fixes #1742.

### CODE REVIEW

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/ProjectSelectorItem.java [hunk 1]
```diff
@@ -30,13 +30,20 @@
...
   private static final Color HOVER_COLOR = UI.getColor("link.foreground");
 
-  ProjectSelectorItem(boolean selected, boolean hovered) {
+  ProjectSelectorItem() {
     setBorder(BorderFactory.createEmptyBorder(2, 15, 2, 0));
     setOpaque(false);
     setHorizontalAlignment(SwingConstants.LEFT);
     setVerticalAlignment(SwingConstants.CENTER);
     setBackground(BACKGROUND_NON_SELECTION_COLOR);
+    // default is non-selected, non-hover state.
```
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/ProjectSelectorItem.java]: nit: I don't think this comment is needed

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/ProjectSelectorItem.java [hunk 2]
```diff
@@ -30,13 +30,20 @@
...
+    setForeground(TEXT_NON_SELECTION_COLOR);
+  }
 
+  /**
+   * Renders the project item hovered (takes precedence), selected or plain.
+   */
+  public void setRenderStyle(boolean selected, boolean hovered) {
```
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/ProjectSelectorItem.java]: package private?

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/ProjectSelectorItem.java (hunk 1 above)
Comment (ivanporty) [reply google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/ProjectSelectorItem.java]: ok, agree.

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/ProjectSelectorItem.java (hunk 2 above)
Comment (ivanporty) [reply google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/ProjectSelectorItem.java]: yes, not used outside of its package now.

**Pull request merged**